src/main/java/com/google/collegeplanner/servlets/SchedulerServlet.java

```
@@ -0,0 +1,140 @@
+
+  /**
+   * Organizes courses from POST request into a given number of semesters
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    JSONObject body;
+    JSONArray selectedClasses;
+    try {
+      body = getPostRequestBody(request);
+      selectedClasses = (JSONArray) body.get("selectedClasses");
+    } catch (NumberFormatException | ClassCastException | ParseException | NullPointerException e) {
+      respondWithError(
+          "Invalid body for POST request.", HttpServletResponse.SC_BAD_REQUEST, response);
+      return;
+    }
+    
+    try {
+      prepareLists(selectedClasses, response);
+    } catch (IOException e ) {
+      return;
+    }
+
+    JSONObject schedules = getSchedules();
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(schedules));
+  }
+
+  /**
+   * This method loads the courses and courseList ArrayLists with correct
+   * information. courses is loaded with the lists of sections for each
+   * course while courseList is loaded with the courseID's of the selected
+   * courses.
+   * @param classes The JSONArray of courseIds of the selected classes
+   * @param response The HttpServletResponse object
+   */
+  private void prepareLists(JSONArray classes, HttpServletResponse response) throws IOException {
+    String courseId;
+    URI uri;
+    JSONArray jsonArray;
+    ApiUtil apiUtil = new ApiUtil();
+
+    courseList = new ArrayList<String>();
+    courses = new ArrayList<ArrayList<Section>>();
+    for (Object obj : classes) {
+      courseId = (String) obj;
+      courseList.add(courseId);
+      try {
+        uri = new URI("https://api.umd.io/v1/courses/" + courseId + "/sections");
+      } catch (URISyntaxException e) {
+        respondWithError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, response);
+        throw new IOException(courseId + " cannot be found in the umd.io database");
```

this isn't really an IOException. What if you just let the URISyntaxException bubble up to the doPost method?